request.go

```diff
@@ -0,0 +1,36 @@
+package httpapi
+
+import (
+	"context"
+	"io"
+	"strings"
+)
+
+type Request struct {
+	ApiBase  string
+	Command  string
+	Args     []string
+	Opts     map[string]string
+	Body     io.Reader
+	Headers  map[string]string
+	DrainOut bool // if set, resp.Close will read all remaining data
```

Comment:
Instead of optionally draining output, what if we introduced a way to _cancel_ requests (separate from `Close`)? As far as I can tell, that's what we're actually using this for.